cocos/base/ccTypes.h

```diff
@@ -165,6 +165,24 @@ struct CC_DLL Color4F
     static const Color4F GRAY;
 };
 
+Color4F& operator+=(Color4F& lhs, const Color4F& rhs);
+Color4F operator+(Color4F lhs, const Color4F& rhs);
+Color4F& operator-=(Color4F& lhs, const Color4F& rhs);
+Color4F operator-(Color4F lhs, const Color4F& rhs);
+
+Color4F& operator*=(Color4F& lhs, const Color4F& rhs);
+Color4F& operator*=(Color4F& lhs, float rhs);
+template<typename T>
+Color4F operator*(Color4F lhs, const T& rhs) {
+    return lhs *= rhs;
+}
```

Comment:
Fair enough. Those template functions saved a few lines of code in the cpp file, but they're probably more confusing than they're worth.  I've updated the PR.